packages/wdio-browserstack-service/tests/service.test.js

```diff
@@ -0,0 +1,244 @@
+    beforeAll(() => {
+        request.get.mockImplementation((url, opts, cb) => cb(
+            null,
+            {statusCode: 200},
+            {
+                automation_session: {
+                    browser_url:
+                        'https://www.browserstack.com/automate/builds/1/sessions/2'
+                }
+            }
+        ));
+        request.put.mockImplementation((url, opts, cb) => {
+            cb(null, {statusCode: 200}, {})
+        })
+    });
+
+    it('should update and get session', () => {
+        return service.onReload(1, 2)
+            .then(() => {
+                expect(request.put).toHaveBeenCalled();
+                expect(request.get).toHaveBeenCalled();
+            });
+    });
+
+    it('should reset failures', () => {
+        const updateSpy = jest.spyOn(service, '_update');
+
+        service.failures = 1;
+        return service.onReload(1, 2)
+            .then(() => {
+                expect(updateSpy).toHaveBeenCalled();
+                expect(service.failures).toEqual(0);
+            })
+    });
+});
+
+describe('_printSessionURL', () => {
+    beforeAll(() => {
+        request.get.mockImplementation((url, opts, cb) => cb(
+            null,
+            {statusCode: 200},
+            {
+                automation_session: {
+                    name: 'Smoke Test',
+                    duration: 65,
+                    os: 'OS X',
+                    os_version: 'Sierra',
+                    browser_version: '61.0',
+                    browser: 'chrome',
+                    device: null,
+                    status: 'failed',
+                    reason: 'CLIENT_STOPPED_SESSION',
+                    browser_url:
+                        'https://www.browserstack.com/automate/builds/1/sessions/2'
+                }
+            }
+        ));
+    });
+    it('should get and log session details', () => {
+        const logInfoSpy = jest.spyOn(log, 'info').mockImplementation((string) => string);
+
+        return service._printSessionURL()
+            .then( () => {
+                expect(logInfoSpy).toHaveBeenCalled();
+                expect(logInfoSpy).toHaveBeenCalledWith(
+                    'OS X Sierra chrome session: https://www.browserstack.com/automate/builds/1/sessions/2'
+                )
+            });
+    });
+
+    it('should throw an error if it recieves a non 200 status code', () => {
+        request.get.mockImplementationOnce((url, opts, cb) => cb(
+            null,
+            {statusCode: 404},
+            {
+            }
+        ));
+
+        expect(service._printSessionURL()).rejects.toEqual('Bad response code: Expected (200), Received (404)!');
+    });
+
+    it('should reject and throw an error if request recieves an error', () => {
+        const e = new Error(`I'm an error!`);
+        request.get.mockImplementationOnce((url, opts, cb) => cb(
+            e,
+            {},
+            {
+            }
+        ));
```

Comment:
no need to put everything into a new line

Reply:
Not sure how that happened, but fixed!